tests/integration/telemetry/stackdriver/stackdriver_filter_test.go

```diff
@@ -96,28 +98,41 @@ func unmarshalFromTemplateFile(file string, out proto.Message, clName string) er
 func TestStackdriverMonitoring(t *testing.T) {
 	framework.NewTest(t).
 		Run(func(ctx framework.TestContext) {
-			sendTraffic(t)
-
-			for _, cl := range ctx.Clusters() {
-				scopes.Framework.Infof("Validating Telemetry for Cluster %v", cl)
-				retry.UntilSuccessOrFail(t, func() error {
-					clName := cl.Name()
-					//Validate cluster names in telemetry below once https://github.com/istio/istio/issues/28125 is fixed.
-					if err := validateMetrics(t, serverRequestCount, clientRequestCount, clName); err != nil {
-						return err
-					}
-					if err := validateLogs(t, serverLogEntry, clName); err != nil {
-						return err
-					}
-					if err := validateTraces(t); err != nil {
-						return err
-					}
-					if err := validateEdges(t, clName); err != nil {
+
+			g, _ := errgroup.WithContext(context.Background())
+			for _, cltInstance := range clt {
+				g.Go(func() error {
+					err := retry.UntilSuccess(func() error {
+						if err := sendTraffic(t, cltInstance); err != nil {
+							return err
+						}
+
+						clName := cltInstance.Config().Cluster.Name()
+						scopes.Framework.Infof("Validating for cluster %s", clName)
+						//Validate cluster names in telemetry below once https://github.com/istio/istio/issues/28125 is fixed.
+						if err := validateMetrics(t, serverRequestCount, clientRequestCount, clName); err != nil {
+							return err
+						}
+						if err := validateLogs(t, serverLogEntry, clName); err != nil {
+							return err
+						}
+						if err := validateTraces(t); err != nil {
+							return err
+						}
+						if err := validateEdges(t, clName); err != nil {
+							return err
+						}
+						return nil
+
+					}, retry.Delay(3*time.Second), retry.Timeout(80*time.Second))
```

Comment:
The the timeout and delay ... should these values be used consistently for telemetry? And if so, should we make them flags so we can vary them based on platform (i.e. maybe we're running on real clusters vs kind)?

Reply:
They at times depend upon the test...

@douglas-reid : what do you think, should we make it a constant?

Reply:
const sounds reasonable to me. i don't think these should vary by platform too much, unless a platform is _really_ CPU constrained or something.  even then, 80 seconds is a long time for retrying

Reply:
Thanks..Fixed